### PR TITLE
tree-comparison demo updates

### PR DESCRIPTION
**Pull request by ChumpChief**
Since we merged #17735 fairly quickly to avoid lagging behind the changing tree API, opening this PR to adjust for any further changes and review feedback.  Please feel free to leave feedback either here or on #17735 and I'll make the updates here :)

### CODE REVIEW

**Review by tyler-cai-microsoft — APPROVED** [review]
LGTM

File: examples/apps/tree-comparison/src/model/legacyTreeInventoryList.ts [hunk 1]
```diff
@@ -143,31 +148,32 @@ export class LegacyTreeInventoryList extends DataObject implements IInventoryLis
...
 					const inventoryItemNodeId = quantityNode.parentage!.parent;
+					const inventoryItemNode =
+						this.tree.currentView.getViewNode(inventoryItemNodeId);
 					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-					this._inventoryItems
-						.get(nodeIdToInventoryItemId(inventoryItemNodeId))!
-						.handleQuantityUpdate(newQuantity);
+					const idNodeId = inventoryItemNode.traits.get("id" as TraitLabel)![0];
+					const idNode = this.tree.currentView.getViewNode(idNodeId);
+					const inventoryItemId = idNode.payload as string;
```
Comment (taylorsw04) [inline examples/apps/tree-comparison/src/model/legacyTreeInventoryList.ts:168]: Why not use the item node's ID directly? You can convert it to UUID form if needed.

Comment (taylorsw04) [reply examples/apps/tree-comparison/src/model/legacyTreeInventoryList.ts:168]: Ditto for all other usages of this uuid-in-payload. It's not clear to me why this is needed, and appears to be redundant.

Comment (ChumpChief) [reply examples/apps/tree-comparison/src/model/legacyTreeInventoryList.ts:168]: I actually started with using the NodeId :) I talk about the change in [this comment](https://github.com/microsoft/FluidFramework/pull/17774/files#diff-9a008d41a408f3f612c0f477705db0af80dac145513001e3f7831a5cee07e522R43) - basically in a real-world scenario I expect the user would be explicitly inputting a unique ID (e.g. a UPC code, model number, etc.), not a uuid.

I think a usage principle should probably be "The NodeId is not part of your schema so don't act like it is".

Comment (taylorsw04) [reply examples/apps/tree-comparison/src/model/legacyTreeInventoryList.ts:168]: Ah, it's like a true item code. Okay, agreed!

**Review by taylorsw04 — APPROVED** [review]

**Review by tyler-cai-microsoft — COMMENTED** [review]
Does the `InventoryListContainerRuntimeFactory` allow you to load from the document correctly? I see that `containerInitializingFirstTime` is implemented but not `containerHasInitialized`

Edit: nevermind, I see where it's loaded. That's a different flow than what I was expecting.

**Comment by ChumpChief** [comment]
> Does the `InventoryListContainerRuntimeFactory` allow you to load from the document correctly? I see that `containerInitializingFirstTime` is implemented but not `containerHasInitialized`
> 
> Edit: nevermind, I see where it's loaded. That's a different flow than what I was expecting.

Yeah, most of our container codes don't use `hasInitialized`/`instantiateFromExisting`.  These are really only useful for ensuring proactive completion of things the container must do before the host receives the `IContainer` reference back from the load.  Otherwise we typically would defer any work until the requestHandler/entryPoint is triggered, so this lifecycle method is usually not needed.

**Pull request merged**